src/components/SendTokens/ETHTokens.js

```diff
@@ -0,0 +1,495 @@
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { TouchableOpacity, Keyboard } from 'react-native';
+import t from 'tcomb-form-native';
+import { utils } from 'ethers';
+import { BigNumber } from 'bignumber.js';
+import styled from 'styled-components/native';
+import debounce from 'lodash.debounce';
+import get from 'lodash.get';
+import isEmpty from 'lodash.isempty';
+import { ethToWei } from '@netgum/utils';
+
+// components
+import { Wrapper } from 'components/Layout';
+import Button from 'components/Button';
+import { TextLink, Label, BaseText } from 'components/Typography';
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import SlideModal from 'components/Modals/SlideModal';
+import SendTokenDetails from 'components/SendTokenDetails';
+
+// utils
+import { formatAmount, formatFiat } from 'utils/common';
+import { fontStyles, spacing, UIColors } from 'utils/variables';
+import { getBalance, getRate, calculateMaxAmount, checkIfEnoughForFee } from 'utils/assets';
+import { makeAmountForm, getAmountFormFields } from 'utils/formHelpers';
+import { checkIfSmartWalletAccount } from 'utils/accounts';
+import { calculateGasEstimate } from 'services/assets';
+import smartWalletService from 'services/smartWallet';
+
+// types
+import type { NavigationScreenProp } from 'react-navigation';
+import type { GasInfo } from 'models/GasInfo';
+import type { TokenTransactionPayload } from 'models/Transaction';
+import type { Balances, Rates } from 'models/Asset';
+import type { RootReducerState } from 'reducers/rootReducer';
+
+// constants
+import { SEND_TOKEN_CONFIRM } from 'constants/navigationConstants';
+import { ETH, defaultFiatCurrency, SPEED_TYPES } from 'constants/assetsConstants';
+
+// actions
+import { fetchGasInfoAction } from 'actions/historyActions';
+import { updateAppSettingsAction } from 'actions/appSettingsActions';
+
+// selectors
+import type { Account } from 'models/Account';
+import Spinner from 'components/Spinner';
+
+const ActionsWrapper = styled.View`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+`;
+
+const SendTokenDetailsValue = styled(BaseText)`
+  ${fontStyles.medium};
+`;
+
+const ButtonWrapper = styled.View`
+  margin-top: ${spacing.rhythm / 2}px;
+  margin-bottom: ${spacing.rhythm + 10}px;
+`;
+
+const Btn = styled(Button)`
+  margin-top: 14px;
+  display: flex;
+  justify-content: space-between;
+`;
+
+const FooterInner = styled.View`
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: flex-end;
+  width: 100%;
+  padding: ${spacing.large}px;
+  background-color: ${UIColors.defaultBackgroundColor};
+`;
+
+const BackgroundWrapper = styled.View`
+  background-color: ${UIColors.defaultBackgroundColor};
+  flexGrow: 1;
+`;
+
+type Props = {
+  assetData: Object,
```

Comment:
Typing? 😏😛

Reply:
Trolling? :D

Reply:
:)) In a friendly manner 😛

Reply:
Haha, I'm glad we're all aware of typing :muscle:. I've previously copied this code and just left what was there since I wasn't sure what type that would be. Now I've added a new type (the `Asset` type doesn't have `token` or `icon`), and updated the rest. Please let me know what you think.

Reply:
It would be good to refactor that `assetData` at some point (in a separate PR). Before we had a one-screen send flow, so it was very easy to construct the final `assetData` object and pass it to the send action.
Now we have multi-steps and multichains form, so I think it's better to pass only really needed info between screens like the token, amount and selected speed/fee, no other fields needed.